onnxruntime/core/mlas/lib/wasm/sgemmc.cpp

```diff
@@ -0,0 +1,824 @@
+
+    ldc - Supplies the first dimension of matrix C.
+
+    alpha - Supplies the scaler multiplier (see SGEMM definition).
+
+Return Value:
+
+    Returns the number of rows handled.
+
+--*/
+{
+    float Row0Block00;
+    float Row0Block01;
+    float Row0Block02;
+    float Row0Block03;
+    float Row0Block10;
+    float Row0Block11;
+    float Row0Block12;
+    float Row0Block13;
+    float Row0Block20;
+    float Row0Block21;
+    float Row0Block22;
+    float Row0Block23;
+    float Row0Block30;
+    float Row0Block31;
+    float Row0Block32;
+    float Row0Block33;
+
+    float Row1Block00;
+    float Row1Block01;
+    float Row1Block02;
+    float Row1Block03;
+    float Row1Block10;
+    float Row1Block11;
+    float Row1Block12;
+    float Row1Block13;
+    float Row1Block20;
+    float Row1Block21;
+    float Row1Block22;
+    float Row1Block23;
+    float Row1Block30;
+    float Row1Block31;
+    float Row1Block32;
+    float Row1Block33;
+
+#if defined(_WIN32)
+
+    if (!ProcessTwoRows) {
+        UNREFERENCED_PARAMETER(lda);
+        UNREFERENCED_PARAMETER(ldc);
+    }
+
+#endif
+
+    do {
+
+        float BElements00;
+        float BElements01;
+        float BElements02;
+        float BElements03;
+        float BElements10;
+        float BElements11;
+        float BElements12;
+        float BElements13;
+        float BElements20;
+        float BElements21;
+        float BElements22;
+        float BElements23;
+        float BElements30;
+        float BElements31;
+        float BElements32;
+        float BElements33;
+
+        float Row0AElements0;
+        float Row0AElements1;
+        float Row1AElements0;
+        float Row1AElements1;
+
+        //
+        // Clear the block accumulators.
+        //
+
+        Row0Block00 = 0.0f;
+        Row0Block01 = 0.0f;
+        Row0Block02 = 0.0f;
+        Row0Block03 = 0.0f;
+        Row0Block10 = 0.0f;
+        Row0Block11 = 0.0f;
+        Row0Block12 = 0.0f;
+        Row0Block13 = 0.0f;
+        Row0Block20 = 0.0f;
+        Row0Block21 = 0.0f;
+        Row0Block22 = 0.0f;
+        Row0Block23 = 0.0f;
+        Row0Block30 = 0.0f;
+        Row0Block31 = 0.0f;
+        Row0Block32 = 0.0f;
+        Row0Block33 = 0.0f;
+
+        if (ProcessTwoRows) {
+            Row1Block00 = 0.0f;
+            Row1Block01 = 0.0f;
+            Row1Block02 = 0.0f;
+            Row1Block03 = 0.0f;
+            Row1Block10 = 0.0f;
+            Row1Block11 = 0.0f;
+            Row1Block12 = 0.0f;
+            Row1Block13 = 0.0f;
+            Row1Block20 = 0.0f;
+            Row1Block21 = 0.0f;
+            Row1Block22 = 0.0f;
+            Row1Block23 = 0.0f;
+            Row1Block30 = 0.0f;
+            Row1Block31 = 0.0f;
+            Row1Block32 = 0.0f;
+            Row1Block33 = 0.0f;
+        }
+
+        //
+        // Compute the 16x1 or 16x2 output block.
+        //
+
+        const float* a = A;
+        size_t k = CountK;
+
+        while (k >= 2) {
```

Comment:
Is there any difference to iterate k by 1? it can get better prefetch with k by 2?

Reply:
it is tricky. better perf on arm currently with k = 2.